File: ports/1.21.1-full/src/main/java/mods/eln/node/INodeEntity.java
package mods.eln.node;

import mods.eln.misc.Direction;
import net.minecraft.client.gui.screens.Screen;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.inventory.AbstractContainerMenu;

import java.io.DataInputStream;

public interface INodeEntity {
    String getNodeUuid();

    void serverPublishUnserialize(DataInputStream stream);

    void serverPacketUnserialize(DataInputStream stream);

    @net.neoforged.api.distmarker.OnlyIn(net.neoforged.api.distmarker.Dist.CLIENT)
    Screen newGuiDraw(Direction side, Player player);

    AbstractContainerMenu newContainer(Direction side, Player player);
}

File: src/main/kotlin/mods/eln/node/INodeEntity.kt
package mods.eln.node

import net.neoforged.api.distmarker.Dist
import net.neoforged.api.distmarker.OnlyIn
import mods.eln.misc.Direction
import net.minecraft.client.gui.screens.Screen
import net.minecraft.world.entity.player.Player
import net.minecraft.world.inventory.AbstractContainerMenu
import java.io.DataInputStream

interface INodeEntity {
    val nodeUuid: String
    fun serverPublishUnserialize(stream: DataInputStream)
    fun serverPacketUnserialize(stream: DataInputStream)

    @OnlyIn(Dist.CLIENT)
    fun newGuiDraw(side: Direction, player: Player): Screen?
    fun newContainer(side: Direction, player: Player): AbstractContainerMenu?
}
